### PR TITLE
Style/immutable

### LINKED ISSUE #109: Use immutable in view

### PR DESCRIPTION
**Pull request by SeqviriouM**
Use everywhere immutable objects in view. Check component to update through function shouldComponentUpdate. Close #109

### CODE REVIEW

**Comment by Nitive** [comment]
Rebased with master

**Comment by Nitive** [comment]
You forgot to correct propTypes

**Comment by Nitive** [comment]
Rebased with master

**Comment by Nitive** [comment]
Rebased with master

**Comment by SeqviriouM** [comment]
Done

File: app/components/ThreadsHeaderTab/index.jsx [hunk 1]
```diff
@@ -13,6 +14,10 @@ export default class ThreadsHeaderTab extends React.Component {
     super(props);
   }
 
+  shouldComponentUpdate(nextProps) {
+    return !Immutable.is(nextProps.isCurrent, this.props.isCurrent);
```
Comment (Nitive) [inline app/components/ThreadsHeaderTab/index.jsx]: Use regular comparison (`===`) for bool type

File: app/components/ThreadsHeader/index.jsx [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
 
   static propTypes = {
-    tabs: PropTypes.arrayOf(React.PropTypes.object),
+    tabs: PropTypes.instanceOf(List),
     currentTab: PropTypes.string,
     changeTab: React.PropTypes.func,
   }
 
+  shouldComponentUpdate(nextProps) {
+    return !Immutable.is(nextProps.currentTab, this.props.currentTab);
```
Comment (Nitive) [inline app/components/ThreadsHeader/index.jsx]: Use regular comparison (`===`) for bool type

**Comment by Nitive** [comment]
++

**Comment by Faraser** [comment]
+1

**Pull request merged**